pkg/controller/nodes/executor.go

```diff
@@ -334,6 +337,15 @@ func (c *nodeExecutor) handleQueuedOrRunningNode(ctx context.Context, nCtx *node
 		return executors.NodeStatusUndefined, err
 	}
 
+	if p.GetPhase() == handler.EPhaseUndefined {
```

Comment:
why can this ever happen, where we get an undefined but no error? That seems like a bug?

Reply:
I don't think it will ever happen. I am not changing the logic.

I believe that check literally means we have a case where phase is not set downstream, and we do not have checks for `handler.EPhaseUndefined` after this.

@surindersinghp @EngHabu if you guys have more context.